config/env/development/database.js

```diff
@@ -0,0 +1,10 @@
+module.exports = ({ env }) => ({
+  connection: {
+    client: 'sqlite',
+    connection: {
+      filename: env('DATABASE_FILENAME', '.tmp/data.db'),
+      schema: false,
+    },
+    useNullAsDefault: true,
+  },
```

Comment:
What is this section of code doing?

Reply:
This is exporting an object to configure the database used by strapi
```
{
  connection: {
    client: 'sqlite',
    connection: {
      filename: env('DATABASE_FILENAME', '.tmp/data.db'),
      schema: false,
    },
    useNullAsDefault: true,
  },
}
```
It says use the sqlite client to connect to the database. The sqlite server writes to a file defined by the environment variable `DATABASE_FILENAME`. If the env var is not provided, use `.tmp/data.db`.